packages/node-core/src/indexer/dictionary.service.test.ts

```diff
@@ -143,4 +205,60 @@ describe('DictionaryService', () => {
     ]);
     expect(dic.batchBlocks[dic.batchBlocks.length - 1]).toBe(333524);
   }, 500000);
+
+  it('able to build queryEntryMap', () => {
+    const dictionaryService = new DictionaryService(DICTIONARY_ENDPOINT, nodeConfig);
+
+    dictionaryService.buildDictionaryEntryMap(mockDS, () => HAPPY_PATH_CONDITIONS);
+    const _map = (dictionaryService as any).mappedDictionaryQueryEntries;
+
+    expect([..._map.keys()]).toStrictEqual(mockDS.map((ds) => ds.startBlock));
+    expect(_map.size).toEqual(mockDS.length);
+  });
+
+  it('able to getDicitonaryQueryEntries', () => {
+    const dictionaryService = new DictionaryService(DICTIONARY_ENDPOINT, nodeConfig);
+    const dictionaryQueryMap = new Map();
+
+    for (let i = 0; i < mockDS.length; i++) {
+      dictionaryQueryMap.set(
+        [mockDS[i].startBlock],
+        HAPPY_PATH_CONDITIONS.filter((dictionaryQuery, index) => i >= index)
```

Comment:
I know you are trying to mock `HAPPY_PATH_CONDITIONS` here, but it is not align with mockDs filters, can we make them match? Or just leave a comment here, otherwise people will confused